File: src/main/java/com/axity/test/product/services/RolService.java
package com.axity.test.product.services;

import com.axity.test.product.repository.RolRepository;
import com.axity.test.product.repository.datamodel.Rol;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.*;
import java.util.List;

@RestController
@RequestMapping("/api/rol")
public class RolService {
    @Autowired
    private RolRepository rolRepository;

    @GetMapping("/getRoles")
    public List<Rol> getRoles() {
        return rolRepository.findAll();
    }
    @PostMapping("/createRol")
    public Rol addRol(@RequestBody Rol rol) {
        return rolRepository.save(rol);
    }
    @PutMapping("/updateRol/{id_rol}")
    public Rol updateRol(@PathVariable int id_rol, @RequestBody Rol rol) {
        return rolRepository.save(rol);
    }

    @DeleteMapping("/deleteRole/{id_rol}")
    public void deleteRol(@PathVariable int id_rol) {
        rolRepository.deleteById(id_rol);
    }

}

File: src/main/resources/static/app/services/rol.service.js
angular.module('productosJM').service('RolService', function($http) {
    this.getRoles = function() {
        return $http.get('/api/rol/getRoles');
    };
    this.createRol = function(rol) {
        return $http.post('/api/rol/createRol', rol);
    };
    this.updateRol = function(id, rol) {
        return $http.put('/api/rol/updateRol/' + id, rol);
    };
    this.deleteRol = function(id) {
        return $http.delete('/api/rol/deleteRol/' + id);
    };
});